draft-parecki-oauth-first-party-native-apps.md

```diff
@@ -443,6 +443,12 @@ In a traditional OAuth deployment using the redirect-based authorization code fl
 Because of these risks, the authorization server MAY decide to require that the user go through a redirect-based flow at any stage of the process based on its own risk assessment.
 
 
+## Credential Attacks {#credential-attacks}
+
+The nature of moving from browser-based authentication to a backend API which is capable of directly receiving user credentials exposes a risk to attacks such as credential stuffing and brute forcing credentails to see if an authorization code is returned by the authorization challenge endpoint. An authorization server may already have a combination of built-in or 3rd party security tools in place to monitor and reduce this risk in the browser-based authentication.  
+
+Implementors SHOULD consider additional security measures to reduce this risk in the authorization challenge endpoint such as monitoringing for suspicious activity. Additionally, the attestation APIs SHOUOLD be used when possible to assert a level of confidence to the authorization server that the request is originating from an application owned by the same party.
```

Comment:
I don't know that "additional" security measures are required but for sure commensurate security measures need to be present. Trust in the client can be obtained by the OS App attestation mechanism.

Reply:
I can tighten up the verbiage. The "commensurate security measures" is what I was getting at.